src/hooks/job-queues/use-submit-job-queue.js

```diff
@@ -0,0 +1,36 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import history from '../../services/history'
+import formatError from '../../services/format-error'
+
+const createMutation = (name) => ({
+    resource: `scheduler/queues/${name}`,
+    type: 'create',
+    data: ({ queue }) => queue,
+})
+
+const useSubmitJobQueue = () => {
+    const engine = useDataEngine()
+    const submitJobQueue = (queue) => {
+        const mutation = createMutation(queue.name)
+        return engine
+            .mutate(mutation, { variables: { queue } })
+            .then(() => {
+                history.push('/')
+            })
+            .catch((error) => {
+                const isValidationError = error.type === 'access'
+
+                // Potential validation error, return it in a format final-form can handle
+                if (isValidationError) {
+                    return formatError(error)
```

Comment:
FYI: when I got a 409 back from the api (due to entering the same name for a job sequence), the formatError function did not work because there was no property `response` on details

Reply:
Hmm, it seems like we should be getting those. Same as we do with jobs. It's my understanding that this type of error reporting hasn't been implemented on all endpoints though, so I can imagine that this hasn't been added in the backend for this endpoint.

@jbee, do you know if the `scheduler/queues/${name}` endpoint should return `errorReports` on submission errors, similar to the `jobConfigurations` endpoint?

Reply:
There is no error report in a response. Each error has a unique error code and message for each possible error.

>   E7020( "Job queue does not exist: `{0}`" ),
>   E7021( "Job queue already exist: `{0}`" ),
>   E7022( "Job `{0}` is already used in another queue: `{1}`" ),
>   E7023( "Job `{0}` is a system job and as such cannot be added to a queue.`" ),
>   E7024( "Job queue must have at least two jobs." ),

Reply:
I see. Is there a reason that this endpoint diverges from the way the other endpoints report back submission errors? From a frontend perspective it would be more maintainable to have a single way for the backend to report back errors. Are we moving the legacy ones to the new pattern as well? Or does it diverge for another reason?

Reply:
For the pattern that @tomzemp reported (trying to create an entity with a name that already exists) the backend will return a response with error reports if it's a job. So for queues we do not get errorReports for this particular pattern.

So if I understand correctly this difference in behaviour between the two endpoints is intentional?

Reply:
So Jan and I discussed this on slack a bit. Basically the gist is that the errorReports we're using for the jobs are just a side-effect of the specific backend operations that endpoint carries out. The new queue endpoints work differently and thus do not get the error reports.

So in short, we'll need to adapt the formatError fn to deal with the new type of error response.